client/src/planwise/client/scenarios/handlers.cljs

```diff
@@ -95,11 +129,10 @@
  :scenarios/get-scenario
  in-scenarios
  (fn [{:keys [db]} [_ id]]
-   {:db  (assoc db :view-state :current-scenario)
-                                        ; reset view state to close any pending dialogs
-    :api (assoc (api/load-scenario id)
+   {:api (assoc (api/load-scenario id)
                 :on-success [:scenarios/save-current-scenario-and-track]
-                :on-failure [:scenarios/scenario-not-found])}))
+                :on-failure [:scenarios/scenario-not-found])
+    :dispatch [:scenarios/reset-state]}))
```

Comment:
Applying my suggestion, this would be:

```
{:api (assoc (api/load-scenario id)
                :on-success [:scenarios/save-current-scenario-and-track]
                :on-failure [:scenarios/scenario-not-found])}))
                :on-failure [:scenarios/scenario-not-found])
 :db (reset-view-state db)}
```